src/relation_engine_server/pull_spec.py

```diff
@@ -1,21 +1,91 @@
 import os
-import subprocess  # nosec
+import requests
+import tarfile
+import tempfile
+import shutil
 
 from . import arango_client, spec_loader
 
 _spec_dir = os.environ.get('SPEC_PATH', '/spec')
+_api_url = 'https://api.github.com/repos/kbase/relation_engine_spec'
+_release_id_path = os.path.join(_spec_dir, '.release_id')
 
 
-def pull_spec():
-    """Download the spec repo to get any updates."""
-    # This always git-pulls no matter what. We may want to throttle or change this in the future.
-    subprocess.check_output(['git', '-C', _spec_dir, 'pull', 'origin', 'master'])  # nosec
-    # Initialize any collections
-    arango_client.init_collections(spec_loader.get_schema_names())
+def download_latest(reset=False, init_collections=True):
+    """Check and download the latest spec and extract it to the spec path."""
+    if reset and os.path.exists(_spec_dir):
+        shutil.rmtree(_spec_dir)
+    os.makedirs(_spec_dir, exist_ok=True)
+    # Download information about the latest release
+    release_resp = requests.get(_api_url + '/releases/latest')
+    release_info = release_resp.json()
+    if release_resp.status_code != 200:
+        # This may be a github API rate usage limit, or some other error
+        return release_info['message']
+    if _has_latest_spec(release_info):
+        return 'already up to date: ' + release_info['tag_name']
+    # Download and extract a new release to /spec/repo
+    spec_repo_path = os.path.join(_spec_dir, 'repo')
+    tarball_url = release_info['tarball_url']
+    resp = requests.get(tarball_url, stream=True)
+    temp_file = tempfile.NamedTemporaryFile()
+    # Download from the tarball url to the temp file
+    _download_file(resp, temp_file.name)
+    # Extract the downloaded tarball into the spec path
+    _extract_tarball(temp_file.name, _spec_dir)
+    temp_file.close()  # Also deletes it
```

Comment:
Why not just use a context manager?

Reply:
Just wanted to reduce nesting for better or worse.. but i'll add a context manager here